source/core/CodalFiber.cpp

```diff
@@ -562,7 +583,7 @@ int codal::invoke(void (*entry_fn)(void *), void *param)
     if (!fiber_scheduler_running())
         return DEVICE_NOT_SUPPORTED;
 
-    if (currentFiber->flags & (DEVICE_FIBER_FLAG_FOB | DEVICE_FIBER_FLAG_PARENT | DEVICE_FIBER_FLAG_CHILD))
+    if (currentFiber->flags & (DEVICE_FIBER_FLAG_FOB | DEVICE_FIBER_FLAG_PARENT | DEVICE_FIBER_FLAG_CHILD) || HAS_THREAD_USER_DATA)
```

Comment:
Could I get a comment here michal? :smile:

Reply:
The OR of flags is what the parametric version of this function does which I assume is correct, since unlike this one it's actually used. 

I'll add comment about thread local